internal/mode/static/state/graph/backend_refs.go

```diff
@@ -127,16 +144,108 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
+	backendTLSPolicy, err := findBackendTLSPolicyForService(
+		backendTLSPolicies,
+		ref,
+		sourceNamespace,
+		configMapResolver,
+	)
+	if err != nil {
+		backendRef = BackendRef{
+			Weight: weight,
+			Valid:  false,
+		}
+
+		cond := staticConds.NewRouteBackendRefUnsupportedValue(err.Error())
+		return backendRef, &cond
+	}
+
 	backendRef = BackendRef{
-		Svc:    svc,
-		Port:   port,
-		Valid:  true,
-		Weight: weight,
+		Svc:              svc,
+		BackendTLSPolicy: backendTLSPolicy,
+		Port:             port,
+		Valid:            true,
+		Weight:           weight,
 	}
 
 	return backendRef, nil
 }
 
+func findBackendTLSPolicyForService(
+	backendTLSPolicies map[types.NamespacedName]*gatewayv1alpha2.BackendTLSPolicy,
+	ref gatewayv1.HTTPBackendRef,
+	routeNamespace string,
+	configMapResolver *configMapResolver,
+) (*gatewayv1alpha2.BackendTLSPolicy, error) {
+	var beTLSPolicy *gatewayv1alpha2.BackendTLSPolicy
+	refNs := routeNamespace
+	if ref.Namespace != nil {
+		refNs = string(*ref.Namespace)
+	}
+
+	for _, btp := range backendTLSPolicies {
```

Comment:
we can have a case of conflicting policies for which we need to resolve as prescribed in https://gateway-api.sigs.k8s.io/guides/api-design/#conflicts